app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java

```diff
@@ -0,0 +1,221 @@
+     *
+     * @param position Position of {@link SMSContact} to delete from contacts list
+     */
+    public void deleteItem(int position) {
+        SMSContact contactToRemove = contacts.get(position);
+        contactManager.removeContact(contactToRemove);
+        contacts.remove(position);
+        filter = new ContactFilter(this, contacts);
+        notifyItemRemoved(position);
+    }
+
+    /**
+     * Update the list of contacts to show
+     * and notify that data have been updated.
+     *
+     * @param newContacts New list of contacts to show and manage
+     */
+    public void updateItems(List<SMSContact> newContacts) {
+        contacts.clear();
+        contacts.addAll(newContacts);
+        filter = new ContactFilter(this, contacts);
+        notifyDataSetChanged();
+    }
+
+
+    //---------------------------- ContactViewHolder ----------------------------
+
+    /**
+     * Class to represent a contact viewed in a row of RecycleView.
+     *
+     * @author Giorgia Bortoletti
+     */
+    public class ContactViewHolder extends RecyclerView.ViewHolder {
+        public TextView contactName;
+        public TextView contactAddress;
+        public ConstraintLayout layout;
+
+        /**
+         * Constructor
+         *
+         * @param itemView
+         */
+        public ContactViewHolder(View itemView) {
+            super(itemView);
+            contactName = itemView.findViewById(R.id.contact_name);
+            contactAddress = itemView.findViewById(R.id.contact_address);
+            layout = itemView.findViewById(R.id.contact_item);
+        }
+
+        /**
+         * Set the item
+         * in its TextView and in its Listener to select one.
+         *
+         * @param contact To set in the item.
+         */
+        public void bind(SMSContact contact, int position) {
+            contactName.setText(contact.getName());
+            contactAddress.setText(contact.getAddress());
+
+            itemView.setOnClickListener(new View.OnClickListener() {
+                /**
+                 * Change status of clicked item.
+                 * The clicked item is the selected contact shown in the MainActivity.
+                 *
+                 * @param view      {@link View} which invoked the click
+                 */
+                @Override
+                public void onClick(View view) {
+                    //layout.setBackgroundColor(ContextCompat.getColor(layout.getContext(), R.color.selectedContact));
+                    int selectedPosition = getAdapterPosition();
+                    SMSContact selectedContact = getItem(selectedPosition);
+                    Intent data = new Intent();
+                    Activity activity = ((Activity) view.getContext());
+
+                    data.putExtra(ActivityConstantsUtils.SELECTED_PHONE_NUMBER, selectedContact.getAddress());
+                    activity.setResult(RESULT_OK, data);
+                    activity.onBackPressed(); //back to home
```

Comment:
Here you can use the `finish()` method, it is more appropriate.
```suggestion
                    activity.finish(); //back to home
```